src/store.ts

```diff
@@ -0,0 +1,26 @@
+import type { Org } from "@app/base/orgs/Org";
+import type { EnsProfile } from "@app/base/registrations/registrar";
+import type { Seed } from "@app/base/seeds/Seed";
+import type { AddressType } from "@app/utils";
+import { get, Writable, writable } from "svelte/store";
+
+export const ens = writable<{ [addressOrName: string]: EnsProfile | null }>({});
+export const type = writable<{ [addressOrName: string]: AddressType }>({});
+export const org = writable<{ [addressOrName: string]: Org }>({});
+export const seed = writable<{ [hostName: string]: Seed }>({});
+
+ens.subscribe(s => console.log("ens store", s));
+type.subscribe(s => console.log("type store", s));
+org.subscribe(s => console.log("org store", s));
+seed.subscribe(s => console.log("seed store", s));
+
+export async function cache<T>(key: string, store: Writable<Record<string, T>>, query: () => Promise<T>): Promise<T> {
```

Comment:
Would be nice if the key was passed as an argument to the closure

Reply:
Yeah good idea 👍